src/index.ts

```diff
@@ -15,18 +15,31 @@ const shell = (cmd: string): string => {
 
 const setAWSCredentials = (
   awsAccessKeyId: string,
-  awsSecretAccessKey: string
+  awsSecretAccessKey: string,
+  awsIamRoleName?: string,
+  awsAccountId?: string
 ) => {
   const creds = `
   [default]
   aws_access_key_id=${awsAccessKeyId}
   aws_secret_access_key=${awsSecretAccessKey}
   `
-  const conf = `
+  let conf = `
   [default]
   region=eu-west-1
   `
 
+  if (awsIamRoleName) {
```

Comment:
You could do something like this
```
const profile =
    awsIamRoleName && awsAccountId
      ? `
  [profile ${awsIamRoleName}]
  region=eu-west-1
  source_profile=default
  role_arn=arn:aws:iam::${awsAccountId}:role/${awsIamRoleName}
  `
      : ""
  const conf = `
  [default]
  region=eu-west-1

  ${profile}
  `
  ```
to get rid of the reassignment.

Reply:
@meraymond2 oh, cool. I'd also thought of creating a separate function which would just append profiles to the config file which would make it easier to test and also maybe support adding multiple profiles if we should ever need that.

What do you think between your suggestion and splitting it out into a separate function?

Reply:
I think split it out, mine ends up looking weird after it's been run through prettier.